SC_redesign/Crop_and_Soil/crop/dormancy.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional
+
+from SC_redesign.Crop_and_Soil.crop.crop_data import CropData, PlantTypes
+
+"""
+This module is based on the "Dormancy" module of SWAT (5:1.2)
+"""
+
+
+class Dormancy:
+    def __init__(self, crop_data: Optional[CropData] = None):
+        self.data = crop_data or CropData
+
+    def go_into_dormancy(self) -> None:
+        """This method performs the actual transition from active to dormant in a crop.
+
+        Details:
+            When method is called, the crop's status is set to dormant and biomass is removed from plant and converted
+            to residue
+        """
+        if self.data.plant_type == PlantTypes.WARM_ANNUAL or PlantTypes.WARM_ANNUAL_LEGUME:
+            # These types of plants do not go into dormancy
+            return
+
+        if self.data.is_dormant:
+            # Crop is already dormant, should not re-perform any dormancy operations on it
+            return
```

Comment:
```suggestion
        no_dormancy = self.data.plant_type == PlantTypes.WARM_ANNUAL or 
          self.data.plant_type == PlantTypes.WARM_ANNUAL_LEGUME or
          self.data.is_dormant
        if no_dormancy:
            return
```
I might be missing paranthesis

Reply:
I think it is better to have these two separate `if` statements because they make it easier to decipher the code, and it is easier to write clear comments for them (IMO). I will rewrite the second comment though because it should include a better reason for why it stops there.